proposals/2024-08-01-Self-Identification.md

```diff
@@ -0,0 +1,110 @@
+
+Note also that due to the recursive structure of schemas, resolving the Schema Object's `$id` keyword can be complex, as each can itself be a relative URI-reference that is resolved against the `$id` in parent schemas.  There is no clear use case for such complexity within other parts of an OpenAPI Description.
+
+### Use Cases
+
+There are several use cases for separating identity and location, including:
+
+* Working around network challenges:
+    * Restrictive network security policies
+    * Intermittent connectivity
+    * High latency / low bandwidth
+    * Document hosts that [require authentication](https://github.com/OAI/OpenAPI-Specification/issues/3270)
+* Abstracting development vs testing vs deployment details
+    * Allowing `.json` and `.yaml` file extensions during development, as is preferred by most IDEs
+    * Using extensions in development and HTTP content negotiation in production
+    * Differing source control repository structure (particularly of shared documents) vs deployment directory and server layouts
+* This separation is necessary (although not, on its own, sufficient) to implement [bundling](https://www.openapis.org/blog/2021/08/23/json-schema-bundling-finally-formalised).
+
+For a more detailed real-world example, see the [OGC example](https://github.com/OAI/sig-moonwalk/discussions/72#user-content-ogc) in the Moonwalk discussion on imports.
+
+Many of these use cases can be worked around, but only by restricting deployment options or performing error-prone reference target rewriting.  Many tools that perform reference rewriting do not take into account the added complexities of referencing in OAS 3.1 compared to 3.0 and earlier.
+
+### Prior Art
+
+Self-identification of resources with identity independent of location is common in the JSON Schema world.  This demonstrates that implementation is not just feasible but well-proven, particularly given that JSON Schema's `$id` is more complex to support than this proposal.
+
+The JSON Schema package used by the OASComply project includes a [schema catalog](https://jschon.readthedocs.io/en/latest/tutorial/catalog.html) with [configurable file and network sources](https://jschon.readthedocs.io/en/latest/examples/file_based_schemas.html) to manage the the URI-to-URL mapping (local files can be considered `file:` URLs).
+
+Self-identification is common in other formats as well.  Notably, the Atom format pioneered the use of [web links with `rel="self"`](https://www.rfc-editor.org/rfc/rfc4287.html#section-4.2.7.2) for this purpose.
+
+## Proposed solution
+
+The proposal is a simplified analog of JSON Schema's `$id` that appears in exactly one place: a new `self` field in the root OpenAPI Object (OAS) and Arazzo Object (Arazzo/Workflows).  When referencing a document that has a `self` field the `self` field SHOULD be used in reference values so that reference values remain the same even if the document is moved.
```

Comment:
Official Naming

```suggestion
The proposal is a simplified analog of JSON Schema's `$id` that appears in exactly one place: a new `self` field in the root OpenAPI Object (OAS) and Arazzo Object (Arazzo). When referencing a document with a `self` field, the `self` field SHOULD be used in reference values so that reference values remain the same even if the document is moved.
```